lib/Core/Executor.cpp

```diff
@@ -905,14 +905,21 @@ Executor::fork(ExecutionState &current, ref<Expr> condition, bool isInternal) {
       }
     }
 
+    // We don't add a constraint explicitly yet
```

Comment:
I agree with @wuestholz. When I was looking at this example initially, I didn't realize that the loop was infinite and couldn't understand why klee wasn't reaching the assertion. So, as a KLEE user, I think that this feature and emitting such a warning would be really helpful.